### PR TITLE
📖 update releasing.md

### PR DESCRIPTION
**Pull request by tuminoid**
BMO releasing.md is badly out of date, update it with more up to date information and commands.

### CODE REVIEW

**Comment by tuminoid** [comment]
/cc @dtantsur @elfosardo @kashifest @furkatgofurov7 

PTAL. I've documented the current process of using tags in main as releases. Release branches etc future improvements are not in scope for this PR.

/unhold

**Review by kashifest — APPROVED** [review]

File: docs/releasing.md [hunk 1]
```diff
@@ -1,38 +1,106 @@
...
+
+BMO release in GitHub will only contain source code as artifact.
+
+### Release notes
+
+Next step is to clean up the release note manually.
+
+- Check for duplicates, reverts, and incorrect classifications of PRs, and
+  whatever release creation tagged to be manually checked.
+- For any superseded PRs (like CAPI uplifts, or commit revertions) that provide
```
Comment (furkatgofurov7) [inline docs/releasing.md:85]: we do not do CAPI uplifts in this repo

File: docs/releasing.md [hunk 2]
```diff
@@ -1,38 +1,106 @@
...
+  no value to the release, create a summary line in `Superseded` section with
+  the PR ids and summary title. This way the changes are acknowledged to be part
+  of the release, but not overwhelming the important changes contained by the
+  release.
+- If the release you're making isn't the latest release (new
+  minor or major, or new patch release in the latest minor release),
+  uncheck the box for latest release.
```
Comment (furkatgofurov7) [inline docs/releasing.md:92]: same comment as in other similar PR: https://github.com/metal3-io/cluster-api-provider-metal3/pull/916#discussion_r1175772191

**Review by furkatgofurov7 — CHANGES_REQUESTED** [review]

File: docs/releasing.md (hunk 1 above)
Comment (tuminoid) [reply docs/releasing.md]: Fixed

File: docs/releasing.md (hunk 2 above)
Comment (tuminoid) [reply docs/releasing.md]: Fixed, is it clearer now?

**Review by furkatgofurov7 — COMMENTED** [review]
Thanks for working on the doc update!

/lgtm

**Comment by tuminoid** [comment]
> Thanks for working on the doc update!

Thanks for the feedback. Can you also approve the changes, you're still requesting changes @furkatgofurov7 ?

**Review by furkatgofurov7 — APPROVED** [review]

**Review by kashifest — APPROVED** [review]

**Pull request merged**